### PR TITLE
[improve][doc]add more explanation for retention configuration

### PR DESCRIPTION
**Pull request by HQebupt**
### Motivation
Current retention configuration is not clear for user especially the explanation of default value . So add more explanation for retention configuration

### Verifying this change

Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

If `yes` was chosen, please highlight the changes

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: (no)
- The schema: (no)
- The default values of configurations: (no)
- The wire protocol: (no)
- The rest endpoints: (no)
- The admin cli options: (no)
- Anything that affects deployment: (no)

### Documentation
Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
- [x] `doc`

### CODE REVIEW

File: conf/broker.conf [hunk 1]
```diff
@@ -1258,10 +1258,12 @@ replicatorPrefix=pulsar.repl
 # due to missing ZooKeeper watch (disable with value 0)
 replicationPolicyCheckDurationSeconds=600
 
-# Default message retention time
+# Default message retention time, less than 0 means never expire.
+# Default is 0, which is to trim ledgers through slowest consumer ledger.
```
Comment (codelipenghui) [inline conf/broker.conf:1262]:
```suggestion
# Default is 0, which means the data will be removed after all the subscriptions are consumed.
```

Comment (codelipenghui) [reply conf/broker.conf:1262]: Just try to provide a more user-perspective description.

File: conf/broker.conf [hunk 2]
```diff
@@ -1258,10 +1258,12 @@ replicatorPrefix=pulsar.repl
 # due to missing ZooKeeper watch (disable with value 0)
 replicationPolicyCheckDurationSeconds=600
 
-# Default message retention time
+# Default message retention time, less than 0 means never expire.
+# Default is 0, which means the data will be removed after all the subscriptions are consumed.
```
Comment (Anonymitaet) [inline conf/broker.conf:1262]:
```suggestion
# Default message retention time.
# The default value is 0, which means the data is removed after all the subscriptions are consumed.
# Value less than 0 means messages never expire.
```

Comment (Anonymitaet) [reply conf/broker.conf:1262]: Same comments for the other occurrences in this PR.

Comment (HQebupt) [reply conf/broker.conf]: Good!

**Review by Technoboy- — APPROVED** [review]

**Comment by Technoboy-** [comment]
@codelipenghui @Anonymitaet Could you help review this again ?

**Review by coderzc — APPROVED** [review]

**Review by codelipenghui — COMMENTED** [review]
For both defaultRetentionTimeInMinutes < 0 and defaultRetentionSizeInMB < 0 means infinite data retention?

We'd better provide more information about the `defaultRetentionTimeInMinutes` and `defaultRetentionSizeInMB` interaction.

File: conf/broker.conf [hunk 3]
```diff
@@ -1261,10 +1261,13 @@ replicatorPrefix=pulsar.repl
...
 
-# Default message retention time
+# Default message retention time.
+# The default value is 0, which means the data is removed after all the subscriptions are consumed.
+# Value less than 0 means messages never expire.
 defaultRetentionTimeInMinutes=0
 
-# Default retention size
+# Default retention size, less than 0 means no infinite size quota.
+# Default is 0, which means the data will be removed after all the subscriptions are consumed.
```
Comment (Anonymitaet) [inline conf/broker.conf:1270]: image

Similar issues still exist.

Comment (HQebupt) [reply conf/broker.conf:1270]: Got it. Thanks

**Comment by HQebupt** [comment]
> For both defaultRetentionTimeInMinutes < 0 and defaultRetentionSizeInMB < 0 means infinite data retention?
> 
> We'd better provide more information about the `defaultRetentionTimeInMinutes` and `defaultRetentionSizeInMB` interaction.

Yes it is. `defaultRetentionTimeInMinutes` and `defaultRetentionSizeInMB` work separately.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**